core/sequencer/sequencer.go

```diff
@@ -297,22 +310,11 @@ func (s *Sequencer) createEpoch(ctx context.Context, d *domain.Domain, msgs []*m
 }
 
 // TODO(gdbelvin): Add leaf at a specific index. trillian#423
-func queueLogLeaf(ctx context.Context, tlog tpb.TrillianLogClient, logID int64, smr *tpb.SignedMapRoot) error {
+func queueLogLeaf(ctx context.Context, logClient *tclient.LogClient, smr *tpb.SignedMapRoot) error {
 	smrJSON, err := json.Marshal(smr)
 	if err != nil {
 		return err
 	}
-	idHash := sha256.Sum256(smrJSON)
-
-	if _, err := tlog.QueueLeaf(ctx, &tpb.QueueLeafRequest{
-		LogId: logID,
-		Leaf: &tpb.LogLeaf{
-			LeafValue:        smrJSON,
-			LeafIdentityHash: idHash[:],
-		},
-	}); err != nil {
-		return fmt.Errorf("trillianLog.QueueLeaf(logID: %v, leaf: %v): %v",
-			logID, smrJSON, err)
-	}
-	return nil
+
+	return logClient.AddLeaf(ctx, smrJSON)
```

Comment:
I'd add a comment here to be explicit that this call blocks until the leaf is actually integrated into the tree.

(Separately, might there be a more descriptive name we could change the function to?)

Reply:
Added comment